py/tests/plane_hog.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/python3
+
+import pykms
+import time
+import random
+
+card = pykms.Card()
+res = pykms.ResourceManager(card)
+conn = res.reserve_connector()
+crtc = res.reserve_crtc(conn)
+
+mode = conn.get_default_mode()
+modeb = mode.to_blob(card)
+
+fb = pykms.DumbFramebuffer(card, mode.hdisplay >> 1, mode.vdisplay >> 1, "AR24");
+pykms.draw_test_pattern(fb);
+
+# Disable request
+card.disable_planes()
+
+plane_list = []
+
+while True:
+     plane = res.reserve_plane(crtc)
+     if plane:
+         print("Got plane %d" % plane.id)
+         plane_list.append(plane)
+     else:
+         break
+
+print("Got %d planes" % len(plane_list))
+
+req = pykms.AtomicReq(card)
+req.add(conn, "CRTC_ID", crtc.id)
+req.add(crtc, {"ACTIVE": 1,
+               "MODE_ID": modeb.id})
+
+input("Press enter to enable crtc %d" % crtc.id)
+r = req.commit_sync(allow_modeset = True)
+
+print("Crtc enable request returned %d\n" % r)
+
+x = 0
+y = 0
+z = 0
+
+for plane in plane_list:
+    input("Press enter to enable plane %d on crtc %d" % (plane.id, crtc.id))
+    req = pykms.AtomicReq(card)
+    req.add(plane, {"FB_ID": fb.id,
+                     "CRTC_ID": crtc.id,
+                     "SRC_X": 0 << 16,
+                     "SRC_Y": 0 << 16,
+                     "SRC_W": fb.width << 16,
+                     "SRC_H": fb.height << 16,
+                     "CRTC_X": x,
+                     "CRTC_Y": y,
+                     "CRTC_W": fb.width,
+                     "CRTC_H": fb.height,
+                     "zorder": z})
+    r = req.commit_sync()
+    print("Plane enable request returned %d\n" % r)
+
+    x = x + 50
+    y = y + 50
+    z = z + 1
+
+conn2 = res.reserve_connector("hdmi")
+crtc2 = res.reserve_crtc(conn2)
+
+mode2 = conn2.get_default_mode()
+modeb2 = mode2.to_blob(card)
+
+req = pykms.AtomicReq(card)
+req.add(conn2, "CRTC_ID", crtc2.id)
+req.add(crtc2, {"ACTIVE": 1,
+                "MODE_ID": modeb2.id})
+
+input("Press enter to turn crtc %d on" % crtc2.id)
+r = req.commit_sync(allow_modeset = True)
+print("Crtc enable request returned %d\n" % r)
+
+x = 0
+y = 0
+z = 0
+
+for plane in reversed(plane_list):
+
+    input("Press enter to disable plane %d on crtc %d" % (plane.id, crtc.id))
+    req = pykms.AtomicReq(card)
```

Comment:
Yep. I initially used ids as the "main" number in kmstest too, but at some point I changed it so that the index is the default number. I.e. kmstest -c 0 shows picture in the first connector, not connector id 0. (@12 would be connector id 12).

Ids can still be used, but I think they're kind of low level things that the user shouldn't normally need to care about on the UI side.